src/sardana/macroserver/msmacromanager.py

```diff
@@ -1243,6 +1245,8 @@ def run(self, params, asynch=True):
             # return self._macro_pointer.getResult()
 
     def _jobEnded(self, *args, **kw):
+        if self.logging_onoff:
```

Comment:
If we remove the handler in `_jobEnded` that mean that in case of synchronous executtion the logger would not be executed. I would move the removal at the end of `runMacro` method.

Reply:
Done